services/user-svc/internal/organization/organization.go

```diff
@@ -367,57 +367,63 @@ func (s ServiceServer) InviteMember(ctx context.Context, req *pb.InviteMemberReq
 	}, nil
 }
 
-func (s ServiceServer) AcceptInvitation(ctx context.Context, req *pb.AcceptInvitationRequest) (*pb.AcceptInvitationResponse, error) {
+func (s ServiceServer) GetInvitationsByOrganization(ctx context.Context, req *pb.GetInvitationsByOrganizationRequest) (*pb.GetInvitationsByOrganizationResponse, error) {
 	db := hwgorm.GetDB(ctx)
 	log := zlog.Ctx(ctx)
 
-	invitationId, err := uuid.Parse(req.InvitationId)
+	claims, err := common.GetAuthClaims(ctx)
 	if err != nil {
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
-	claims, err := common.GetAuthClaims(ctx)
+	organizationID, err := uuid.Parse(req.OrganizationId)
 	if err != nil {
-		return nil, err
+		return nil, status.Error(codes.Internal, err.Error())
 	}
 
-	// Check if invite exists
-	currentInvitation, err := GetInvitationByIdAndEmail(db, claims.Email, invitationId)
+	userID, err := common.GetUserID(ctx)
 	if err != nil {
-		if hwgorm.IsOurFault(err) {
-			return nil, status.Error(codes.Internal, err.Error())
-		} else {
-			return nil, status.Error(codes.InvalidArgument, err.Error())
-		}
+		return nil, status.Error(codes.Internal, err.Error())
 	}
 
-	if currentInvitation.State == pb.InvitationState_INVITATION_STATE_ACCEPTED {
-		return &pb.AcceptInvitationResponse{}, nil
-	} else if currentInvitation.State != pb.InvitationState_INVITATION_STATE_PENDING {
-		return nil, status.Error(codes.InvalidArgument, "only pending invitations can be accepted")
+	hasAccess, err := IsInOrganization(db, organizationID, userID)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
 	}
 
-	// Update invitation state
-	invitation := Invitation{
-		ID: invitationId,
+	if !hasAccess || hwutil.Contains(claims.Organizations, organizationID) {
```

Comment:
We don't need to do this membership check twice. The result of `hasAccess` comes from the database. `hwutil.Contains(claims.Organizations, organizationID)` comes from ID token. We rely on and prefer the later one.

Reply:
yes, but do we update the token every time the user joins a new organization, otherwise the user could be part of the organization, but cannot access the data

Reply:
> but do we update the token every time the user joins a new organization
Yes, after while. When the access token expires.

Correct, but this is our consistency model for the access token. When the request hits that method, [the `hwutil.Contains(claims.Organizations, organizationID)` check was already performed](https://github.com/helpwave/services/blob/main/libs/common/grpc.go#L164) with the OrganizationID of the `X-Organization` metadata. The user is likely to call this method for the current organization. Therefore, the user is already a part of the organization and the later condition will always be true.